synapse/storage/databases/main/room_batch.py

```diff
@@ -18,7 +18,7 @@
 
 
 class RoomBatchStore(SQLBaseStore):
-    async def get_insertion_event_by_batch_id(
+    async def get_insertion_event_id_by_batch_id(
         self, room_id: str, batch_id: str
     ) -> Optional[str]:
         """Retrieve a insertion event ID.
```

Comment:
TODO: Fix up this failing Sytest test, https://github.com/matrix-org/synapse/runs/4089675723?check_suite_focus=true

[`Backfill checks the events requested belong to the room`](https://github.com/matrix-org/sytest/blob/48be0670c7aae119952a0e2ed3434bc5c6afcc2f/tests/50federation/34room-backfill.pl#L201-L254)

Reply:
Fixed via https://github.com/matrix-org/synapse/pull/11114/commits/df2a1523fcecc0f70aeed5e9c4c6bbb51cc06239

```
$ POSTGRES=1 ./run-tests.pl --synapse-directory /Users/eric/Documents/github/element/synapse tests/50federation/34room-backfill.pl --server-log --client-log
```

`sytest` -> `server-0/homeserver.log`
```
2021-12-16 17:48:24,350 - synapse.http.server - 98 - ERROR - GET-13 - Failed handle request via 'FederationBackfillServlet': <SynapseRequest at 0x10bfbed60 method='GET' uri='/_matrix/federation/v1/backfill/!AZJMOtEzYKmrYllmEA:localhost:8800?limit=1&v=%243hYFMeqCY38Ak7o7nbNpGPWGoPjHbHhxCxVnQWaf958' clientproto='HTTP/1.1' site='8800'>
Traceback (most recent call last):
  File "/Users/eric/Documents/github/element/synapse/synapse/http/server.py", line 267, in _async_render_wrapper
    callback_return = await self._async_render(request)
  File "/Users/eric/Documents/github/element/synapse/synapse/http/server.py", line 467, in _async_render
    callback_return = await raw_callback_return
  File "/Users/eric/Documents/github/element/synapse/synapse/federation/transport/server/_base.py", line 315, in new_func
    response = await func(
  File "/Users/eric/Documents/github/element/synapse/synapse/federation/transport/server/federation.py", line 174, in on_GET
    return await self.handler.on_backfill_request(origin, room_id, versions, limit)
  File "/Users/eric/Documents/github/element/synapse/synapse/federation/federation_server.py", line 195, in on_backfill_request
    pdus = await self.handler.on_backfill_request(
  File "/Users/eric/Documents/github/element/synapse/synapse/handlers/federation.py", line 1059, in on_backfill_request
    events = await self.store.get_backfill_events(room_id, pdu_list, limit)
  File "/Users/eric/Documents/github/element/synapse/synapse/storage/databases/main/event_federation.py", line 1120, in get_backfill_events
    event_ids = await self.db_pool.runInteraction(
  File "/Users/eric/Documents/github/element/synapse/synapse/storage/database.py", line 741, in runInteraction
    result = await self.runWithConnection(
  File "/Users/eric/Documents/github/element/synapse/synapse/storage/database.py", line 846, in runWithConnection
    return await make_deferred_yieldable(
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/python/threadpool.py", line 238, in inContext
    result = inContext.theWork()  # type: ignore[attr-defined]
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/python/threadpool.py", line 254, in <lambda>
    inContext.theWork = lambda: context.call(  # type: ignore[attr-defined]
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/python/context.py", line 118, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/python/context.py", line 83, in callWithContext
    return func(*args, **kw)
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/enterprise/adbapi.py", line 293, in _runWithConnection
    compat.reraise(excValue, excTraceback)
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/python/deprecate.py", line 298, in deprecatedFunction
    return function(*args, **kwargs)
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/python/compat.py", line 404, in reraise
    raise exception.with_traceback(traceback)
  File "/Users/eric/Documents/github/element/synapse/env/lib/python3.9/site-packages/twisted/enterprise/adbapi.py", line 284, in _runWithConnection
    result = func(conn, *args, **kw)
  File "/Users/eric/Documents/github/element/synapse/synapse/storage/database.py", line 841, in inner_func
    return func(db_conn, *args, **kwargs)
  File "/Users/eric/Documents/github/element/synapse/synapse/storage/database.py", line 609, in new_transaction
    r = func(cursor, *args, **kwargs)
  File "/Users/eric/Documents/github/element/synapse/synapse/storage/databases/main/event_federation.py", line 1172, in _get_backfill_events
    event_lookup_result["depth"],
TypeError: 'NoneType' object is not subscriptable
```